### DISCUSSION TITLE
AquaBEHER 1.4.0

### DISCUSSION THREAD

**Discussion by RobelTakele** [RobelTakele/AquaBEHER#10] [Q&A] [answered]
# AquaBEHER 1.4.0

### 🆕 Added

- **🛠 Parameter Validation**: An improved and more robust version of the
  parameter validation checks, along with additional data quality and
  validation checks for the `seasFcstQBR` function.

  - **🔍 Input Validation**: Checks added for `hisYearStart`,
    `hisYearEnd`, and the structure of `seasRain` and `hisWSCvar`.

  - **❓ Null and Missing Checks**: Added `is.null` and `is.na` checks
    for `rainTerc` and key data columns.

  - **⚠ Unrealistic Data Warning**: Warnings for unrealistic `hisWSCvar`
    values (e.g., values outside 0-365 days) and negative rainfall
    values.

  - **📅 Year Filtering**: Ensures `seasRain` and `hisWSC` match on
    years after filtering.

- **🧪 Testing Structure**: Established testing structure using
  `testthat` for the `seasFcstQBR` function.

- **📖 Documentation**: Added a Seasonal Forecast section to the
  vignette.

### 🔄 Changed

- **📚 Documentation**: Improved documentation for `seasFcstQBR`.

- **🔄 Function Name**: Changed the function name from `seasFcst` to
  `seasFcstQBR`.

- **📝 DESCRIPTION File**: Edited the DESCRIPTION file as directed by
  CRAN.

<hr /><em>This discussion was created from the release <a href='https://github.com/RobelTakele/AquaBEHER/releases/tag/v1.4.0'>AquaBEHER 1.4.0</a>.</em>

**Comment by gtemesgen** [comment]
How can we get the input for the seasonal forecast of rainfall ?

**Comment by RobelTakele** [comment]
One way to obtain seasonal forecast of rainfall in the form of categorical tercile forecasts is from national meteorological offices, get the forecast and import it as a dataframe.

eg. for the seasonal rainfall forecast with tercile probability of Below Normal = 15%, Normal = 10% and Above Normal = 75%; you can import it as:
rainTerc <- data.frame(T1 = 0.15, T2 = 0.10, T3 = 0.75)

**Comment by RobelTakele** [comment, marked as answer]
One way to obtain seasonal forecast of rainfall in the form of categorical tercile forecasts is from national meteorological offices, get the forecast and import it as a dataframe.

eg. for the seasonal rainfall forecast with tercile probability of Below Normal = 15%, Normal = 10% and Above Normal = 75%; you can import it as:
rainTerc <- data.frame(T1 = 0.15, T2 = 0.10, T3 = 0.75)
